leakcanary-android-utils/src/main/java/leakcanary/DumpWrapper.kt

```diff
@@ -0,0 +1,42 @@
+package leakcanary
+
+import android.os.Build
+import android.os.Debug
+import android.util.Log
+import leakcanary.internal.checkNotMainThread
+import java.io.IOException
+import java.lang.Exception
+
+object DumpWrapper {
+  private const val TAG = "DumpWrapper"
+  private const val ANDROID_R = Build.VERSION_CODES.Q + 1
+  private val VERSION_MATCH = Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP && Build
+    .VERSION.SDK_INT <= ANDROID_R
+
+  @Throws(IOException::class)
+  fun dumpHprofData(fileName: String) {
+    if (isSupportFastDump()) {
+      checkNotMainThread()
+      try {
+        fastDumpClass?.getDeclaredMethod("forkDumpHprof", String::class.java)?.invoke(null, fileName)
+      } catch (e: Exception) {
+        e.printStackTrace()
+      }
+    } else {
+      Debug.dumpHprofData(fileName)
+    }
+  }
+
+  private val fastDumpClass by lazy {
+    try {
+      return@lazy Class.forName("com.squareup.leakcanary.FastDump")
```

Comment:
Yep, this is good, it means fast dump is optional and you have to specifically include the artifact to get it.